src/rpc/server.cpp

```diff
@@ -63,6 +64,34 @@ RPCServer::InterruptedListener::InterruptedListener(std::function<void ()> callb
 }
 RPCServer::InterruptedListener::~InterruptedListener() {}
 
+
+RPCServer::BlockChangeBlocker::BlockChangeBlocker() : m_interrupted_listener([this] {
+            m_cv.notify_all();
+        })
+{
+    {
+        LOCK(cs_main);
+        m_last_block_hash = chainActive.Tip()->GetBlockHash();
```

Comment:
In commit "Add RPC utility to block until tip changes or RPC is interrupted"

Do you need to acquire m_cs while updating m_last_block variables here? It doesn't seem like there is a guard against these writes being delayed and UpdatedBlockTip trying to update the values at the same time. Or if this is safe, should add a comment here saying why m_cs isn't needed.